`sphinx` errors in `make doc`

Following the release of `sphinx==5.1.0` to `pypi` on July 24, 2022, `make doc` is failing during the CI. We have confirmed that it runs successfully for `sphinx==5.0.2`.

We need to review the updates from `5.0` to `5.1` and determine if we will be making changes to accommodate the new version. For the time being PR #1624 has updated the requirement of `sphinx` to `sphinx==5.0.2`.

I did some digging on this and it appears that a patch has been released for `5.1.0`, as there is now `sphinx==5.1.1` available. I installed this on my machine and was able to run `make doc` successfully. Additionally, I confirmed that `5.1.0` still produces the same error as before.

```
Handler <function _process_docstring at 0x12fc30d30> for event 'autodoc-process-docstring' threw an exception (exception: pop from an empty deque)
make[1]: *** [html] Error 2
make: *** [/Users/ethandebandi/Documents/git/arkouda//docs/index.html] Error 2
```

--- a/arkouda-env-dev.yml
+++ b/arkouda-env-dev.yml
@@ -21,7 +21,7 @@ dependencies:
   - pexpect
   - pytest>=6.0
   - pytest-env
-  - Sphinx==5.0.2
+  - Sphinx==5.1.1
   - sphinx-argparse
   - sphinx-autoapi
   - typed-ast

--- a/arkouda-env-dev.yml
+++ b/arkouda-env-dev.yml
@@ -21,7 +21,7 @@ dependencies:
   - pexpect
   - pytest>=6.0
   - pytest-env
-  - Sphinx==5.1.1
+  - Sphinx>=5.1.1
   - sphinx-argparse
   - sphinx-autoapi
   - typed-ast

--- a/pydoc/requirements.txt
+++ b/pydoc/requirements.txt
@@ -18,7 +18,7 @@ pyarrow>=1.0.1
 pexpect
 pytest>=6.0
 pytest-env
-Sphinx==5.0.2
+Sphinx>=5.1.1
 sphinx-argparse
 sphinx-autoapi
 typed-ast

--- a/setup.py
+++ b/setup.py
@@ -156,7 +156,7 @@ setup(
     # projects.
     extras_require={  # Optional
         'dev': ['pexpect', 'pytest>=6.0', 'pytest-env',
-                'Sphinx==5.0.2', 'sphinx-argparse', 'sphinx-autoapi',
+                'Sphinx>=5.1.1', 'sphinx-argparse', 'sphinx-autoapi',
                 'mypy>=0.931', 'typed-ast', 'black', 'isort',
                 'flake8'],
     },

